scheduler/src/cook/rest/api.clj

```diff
@@ -3052,6 +3052,103 @@
        :post-enacted? (constantly false)
        :post! post-shutdown-leader!})))
 
+;;
+;; /compute-clusters
+;;
+
+(def InsertComputeClusterRequest
+  {; IP address / URL of the cluster
+   :base-path s/Str
+   ; base-64-encoded certificate
+   :ca-cert s/Str
+   ; Unique name of the cluster
+   :name s/Str
+   ; State of the cluster
+   :state (s/enum
+            ; Jobs will get scheduled on this cluster
+            "running"
+            ; No new jobs / synthetic pods will
+            ; get scheduled on this cluster
+            "draining"
+            ; The cluster is gone
+            "deleted")
+   ; The state can't be modified when it's locked
+   (s/optional-key :state-locked?) s/Bool
+   ; Key used to look up the configuration template
+   :template s/Str})
+
+(defn create-compute-cluster!
+  [conn ctx]
+  (let [result (cc/update-compute-clusters conn (-> ctx :request :body-params) nil true)]
+    ; TODO replace with real result handling with error checks
+    (log/info "Result of create-compute-cluster! REST API call"
+              {:input (-> ctx :request :body-params)
+               :result result})))
+
+(defn compute-cluster-exists?
+  [db name]
+  ;TODO might not need this. this is checked in the main update function
+  false)
+
+(defn check-compute-cluster-conflict
+  [conn ctx]
+  (let [db (d/db conn)
+        name (-> ctx :request :body-params :name)
+        exists? (compute-cluster-exists? db name)]
+    (if exists?
+      [true {::error (str "Compute cluster with name " name " already exists")}]
+      false)))
+
+(defn check-compute-cluster-allowed
+  [is-authorized-fn ctx]
+  (let [request-user (get-in ctx [:request :authorization/user])
+        impersonator (get-in ctx [:request :authorization/impersonator])
+        request-method (get-in ctx [:request :request-method])]
+    (if-not (is-authorized-fn request-user
+                              request-method
+                              impersonator
+                              {:owner ::system :item :compute-clusters})
+      [false {::error
+              (str "You are not authorized to access compute cluster information")}]
+      true)))
+
+(defn base-compute-cluster-handler
+  [is-authorized-fn leadership-atom leader-selector resource-attrs]
+  (base-cook-handler
+    (merge
+      ;; only the leader handles compute-cluster requests
+      (redirect-to-leader leadership-atom leader-selector)
+      {:allowed? (partial check-compute-cluster-allowed is-authorized-fn)
+       :existed? (constantly true)
+       ;; triggers path for moved-temporarily?
+       :exists? (fn [_] @leadership-atom)}
```

Comment:
These can be removed since they're defined exactly as such in `redirect-to-leader`

Reply:
done